pkg/apis/core/validation/shoot.go

```diff
@@ -1029,6 +1038,88 @@ func validateKubeletConfigEvictionSoftGracePeriod(eviction *core.KubeletConfigEv
 	return allErrs
 }
 
+// Reserved PIDs have to be non-negative
+const ReservedPIDsMinimum int64 = 0
+
+func validateKubeletConfigKubeReserved(reserved *core.KubeletConfigKubeReserved, fldPath *field.Path) field.ErrorList {
```

Comment:
WDYT about introducing checks based on the machine type? 
Enforcing some sane defaults, so that Shoots cannot be created which nodes are garuanteed to not be healthy.
For instance reserving more kubelet, system CPU,Memory, EphermalStorage than available according to the machine type.

Basically making it harder for the Shoot creator to "shoot" himself in the foot.

Reply:
As I have also written to you in pn, I think we should be clear which cases of misconfiguration we can be verified by Gardener and which not. And lets also determine if it even makes sense to have validation if there are so many cases that we cannot validate anyways. That I think is also the reason why other Kubernetes Services do not allow configuring these options. 
- cGroup names cannot be validated by Gardener upfront (user might have prepared the Os image in a way that Gardener does not know about). However we could say that the kubelet options `kubeReservedCgroup` and `systemReservedCgroup`  are not exposed and are always the Gardener created cGroups.
- `Kube-reserved` depends on the load on the node (workload dependent) and `system-reserved` depends on the OS. Here we can only validate against edge cases (request more memory than available on the node). Is that validation then useful?

I guess it also depends on the use case that you have in mind. 

Inviting @timuthy @rfranzke  @tim-ebert

Reply:
Hm, I tend to not over-complicate this PR for now. It's an advanced setting (like many other fields that we already have) and having such sophisticated validation will make it harder for us at the moment. I appreciate your thoughts about it, but I'd probably start small and unblock users that need to configure these settings. If we see that it's misused then we can revisit the validation topic. WDYT?